java/java.hints/test/unit/src/org/netbeans/modules/java/hints/infrastructure/ErrorHintsTestBase.java

```diff
@@ -335,8 +335,8 @@ protected final int positionForErrors() throws IllegalStateException {
         Diagnostic<?> found = null;
         for (Diagnostic<?> d : info.getDiagnostics()) {
             if (d.getKind() == Diagnostic.Kind.ERROR && (supportedErrorKeys == null || supportedErrorKeys.contains(d.getCode()))) {
-                if (found == null) {
-                    found = d;
+                if (found == null || found.getCode().equals(d.getCode())) {
```

Comment:
Same errors show multiple times in compound var declaration. To handle the multiple errors this check is require.

Reply:
Position of multiple error will be the same or different.
If position is same then no need to get the position of the last error, one can simply break from the loop

Reply:
This is require to check the more than one errors with different types.

Reply:
Why twice null check

Reply:
Second null check "if(found == null)" require to assign the first error and bypass the next error of with same error code.

Reply:
But why put both checks in same if, we can avoid duplicate null check.